packages/web3-validator/src/utils.ts

```diff
@@ -218,6 +218,7 @@ export const transformJsonDataToAbiFormat = (
 			abiName = abi.name;
 			abiComponents = abi.components as FullValidationSchema;
 			// If its short form string value e.g. ['uint']
+			// todo abis is FullValidationSchema aka ReadOnlyArray<AbiParameter>, thus abi cannot be something else, so may next two ifs are reduntant
```

Comment:
Next two if are checking internal members of `abis` that is `FullValidationSchema`, and not `abis` param it self.